### PR TITLE
ROX-26093: Improve sensor behavior when using wrong init bundle

### PR DESCRIPTION
**Pull request by jschnath**
### Description

Turns the warning into an error and makes it much more visible in the log. Style similar to other ASCII demarcations in logs.

### User-facing documentation

- [x] CHANGELOG is updated **OR** update is not needed
- [x] [documentation PR](https://spaces.redhat.com/display/StackRox/Submitting+a+User+Documentation+Pull+Request) is created and is linked above **OR** is not needed

### Testing and quality

- [x] the change is production ready: the change is GA or otherwise the functionality is gated by a feature flag
- [x] CI results are inspected

#### Automated testing

- [ ] added unit tests
- [ ] added e2e tests
- [ ] added regression tests
- [ ] added compatibility tests
- [ ] modified existing tests

#### How I validated my change

Manually tested that the error appears and confirmed that it looks good in the logs.
Screenshots of how this looks in k9s / bash
image
image

### CODE REVIEW

**Review by lvalerom — COMMENTED** [review]
Looks good. Did you manage to reproduce the error and confirm that the message is printed properly?

File: sensor/common/centralclient/client.go [hunk 1]
```diff
@@ -315,7 +315,9 @@ func RemoteCertLoader(httpClient *Client) CertLoader {
 	return func() []*x509.Certificate {
 		certs, err := httpClient.GetTLSTrustedCerts(context.Background())
 		if err != nil {
-			log.Warnf("Error fetching centrals TLS certs: %s", err)
+			log.Errorf("\n#------------------------------------------------------------------------------\n"+
+				"# Error fetching centrals TLS certs: %s\n"+
```
[the remark this answers is not in the document]
Comment (jschnath) [reply sensor/common/centralclient/client.go]: Done 👍

**Review by vikin91 — COMMENTED** [review]
The code change looks good!

> Style similar to other ASCII demarcations in logs.

Where did you get the inspiration from? I tried to search for `------` in our code but could not find anything. If there are no examples in our code, maybe we could consider using `=` instead of `-`? Showing a screenshot with both somewhere in Slack could lead to a voting 😉   It is not a blocker, I am just curious.

> Manually tested that the error appears and confirmed that it looks good in the logs.

Writing this and not providing a screenshot is a crime 😂

**Comment by jschnath** [comment]
>Where did you get the inspiration from?

I searched for "---" and "___" demarcations and found this style used in a bunch of configuration files. I think it looks fine in a log too.

> Writing this and not providing a screenshot is a crime 😂
Next time I'll do that 😅 don't want to set this up again right now

image

**Comment by vikin91** [comment]
> > Where did you get the inspiration from?
> 
> I searched for "---" and "___" demarcations and found this style used in a bunch of configuration files. I think it looks fine in a log too.
> 
> > Writing this and not providing a screenshot is a crime 😂
> > Next time I'll do that 😅 don't want to set this up again right now
> 
> image

Jan, this is a comment in the code, not a log line. That is a difference.

**Review by vikin91 — COMMENTED** [review]
The change looks good, but I would like to request to:
1. Update the checkboxes in the PR description
2. Add a screenshot to the PR description with the log output introduced in the PR

**Review by lvalerom — COMMENTED** [review]
Looks good to me but I would request the same things Piotr requested before approving.

**Comment by jschnath** [comment]
Screenshots of how this looks in k9s / bash
image
image

**Comment by jschnath** [comment]
> Should we open this style to voting in a wider channel? 

I don't personally think that this is such a big decision to require voting from the wider team, but if you both think that way I may be wrong.

> or maybe add a helper function for this kind of messages so we have consistency across the code base?

That sounds like a good option. But where should it go, inside the logger? I just had a look and our Logger class uses a `zap.SugaredLogger` internally, I could add a function to the logger that adds lines before and after a string. Or would this be a better fit for some string manipulation package?

**Review by lvalerom — APPROVED** [review]
> That sounds like a good option. But where should it go, inside the logger? I just had a look and our Logger class uses a zap.SugaredLogger internally, I could add a function to the logger that adds lines before and after a string. Or would this be a better fit for some string manipulation package?

We can think about it in a follow-up. I wouldn't block merging this for that 🙂

**Pull request merged**